src/Symfony/Component/DomCrawler/Form.php

```diff
@@ -197,9 +197,21 @@ public function getUri()
     {
         $uri = parent::getUri();
 
-        if (!in_array($this->getMethod(), array('POST', 'PUT', 'DELETE', 'PATCH')) && $queryString = http_build_query($this->getValues(), null, '&')) {
-            $sep = false === strpos($uri, '?') ? '?' : '&';
-            $uri .= $sep.$queryString;
+        if (!in_array($this->getMethod(), array('POST', 'PUT', 'DELETE', 'PATCH'))) {
+            $query = parse_url($uri, PHP_URL_QUERY);
+            $currentParameters = array();
+            if ($query) {
+                foreach (explode('&', $query) as $param) {
```

Comment:
Could it be possible to replace these two combined `explode` functions + `foreach` by a call to the PHP [parse_str() function](http://php.net/parse-str)?